lib/Models/WebFeatureServiceCatalogItem.ts

```diff
@@ -0,0 +1,486 @@
+        )
+      )
+    ) {
+      throw new TerriaError({
+        sender: this,
+        title: i18next.t(
+          "models.webFeatureServiceCatalogItem.unsupportedGeojsonOutputTitle"
+        ),
+        message: i18next.t(
+          "models.webFeatureServiceCatalogItem.unsupportedGeojsonOutputMessage",
+          this
+        )
+      });
+    }
+
+    const url = this.uri
+      .clone()
+      .setSearch(
+        combine(
+          {
+            service: "WFS",
+            request: "GetFeature",
+            typeName: this.typeNames,
+            version: "1.1.0",
+            outputFormat: "JSON",
+            srsName: "EPSG:4326",
+            maxFeatures: this.maxFeatures
+          },
+          this.parameters
+        )
+      )
+      .toString();
+
+    const getFeatureResponse = await loadText(proxyCatalogItemUrl(this, url));
+
+    // If request returns XML, try to find error message
+    if (getFeatureResponse.startsWith("<")) {
+      try {
+        const error = xml2json(getFeatureResponse);
+
+        throw new TerriaError({
+          sender: this,
+          title: i18next.t(
+            "models.webFeatureServiceCatalogItem.missingDataTitle"
+          ),
+          message: `${i18next.t(
+            "models.webFeatureServiceCatalogItem.missingDataMessage",
+            this
+          )} <br/>Error: ${error.ExceptionReport?.Exception?.ExceptionText ||
+            error.toString()}`
+        });
+      } catch (e) {
+        console.log(getFeatureResponse);
+        throw new TerriaError({
+          sender: this,
+          title: i18next.t(
+            "models.webFeatureServiceCatalogItem.missingDataTitle"
+          ),
+          message: `${i18next.t(
+            "models.webFeatureServiceCatalogItem.missingDataMessage",
+            this
+          )}`
+        });
+      }
+    }
+
+    runInAction(() => {
+      this.geojsonCatalogItem = new GeoJsonCatalogItem(
+        createGuid(),
+        this.terria
+      );
+
+      this.geojsonCatalogItem.setTrait(
+        CommonStrata.definition,
+        "geoJsonData",
+        JSON.parse(getFeatureResponse)
+      );
+
+      if (isDefined(this.style))
+        this.geojsonCatalogItem.setTrait(
+          CommonStrata.definition,
+          "style",
+          this.style
+        );
+    });
+
+    await this.geojsonCatalogItem!.loadMapItems();
+  }
+
+  @computed
+  get typeNamesArray(): ReadonlyArray<string> {
+    if (Array.isArray(this.typeNames)) {
+      return this.typeNames;
+    } else if (this.typeNames) {
+      return this.typeNames.split(",");
+    } else {
+      return [];
+    }
+  }
+
+  @computed
+  get canExportData() {
+    return isDefined(this.geojsonCatalogItem?.geoJsonData);
+  }
+
+  async exportData() {
+    if (isDefined(this.geojsonCatalogItem?.geoJsonData)) {
+      return {
+        name: `${this.name} export.json`,
+        file: new Blob([this.geojsonCatalogItem!.geoJsonData.toString()])
```

Comment:
Perhaps `json.stringify` rather than `.toString()` to fix the export

Reply:
Yep good point! thanks